packages/firestore/src/local/local_store.ts

```diff
@@ -332,26 +332,24 @@ export class LocalStore {
     );
   }
 
-  /** Returns the local view of all documents in a mutation batch. */
-  lookupLocalWrite(batchId: BatchId): Promise<LocalWriteResult | null> {
-    return this.persistence.runTransaction('Lookup batch', false, txn => {
-      let batch: MutationBatch;
-      return this.mutationQueue
-        .lookupMutationBatch(txn, batchId)
-        .next(promisedBatch => {
-          if (promisedBatch) {
-            batch = promisedBatch;
-            const keys = batch.keys();
-            return this.localDocuments
-              .getDocuments(txn, keys)
-              .next((changedDocuments: MaybeDocumentMap) => {
-                return { batchId: batch.batchId, changes: changedDocuments };
-              });
-          } else {
-            return PersistencePromise.resolve(null);
-          }
-        });
-    });
+  /** Returns the local view of the documents affected by a mutation batch. */
+  // PORTING NOTE: Multi-tab only.
+  lookupMutationDocuments(batchId: BatchId): Promise<MaybeDocumentMap | null> {
```

Comment:
This method seems a bit odd to me. The name doesn't suggest to me that it looks up the local view of the documents (rather than just returning the list of documents in the mutation or something).  I sort of want to rename it to lookupMutatedDocuments() but that's not right because sometimes it is run against documents that have a mutation and sometimes it's run against documents that previously had a mutation.  So I can't come up with a great name...

It's also a bit odd that it may or may not succeed (it can return null if it fails to get the list of document keys).

Part of me is tempted to just inline the method where it's used since it's only used once and it's difficult to name / describe it well.  But I don't feel strongly that that's an improvement either.  So feel free to leave it as-is unless my mild griping has triggered some inspiration on your side. :)

Reply:
I can't really come up with a name that alleviates all of these concerns. I think `lookupMutatedDocuments` might be slightly better, but I am also not 100% convinced. For now, i decided to keep it as is.

I would also rather not inline it as that would move the transaction logic into SyncEngine.